lib/framework/procbot.ts

```diff
@@ -108,58 +118,52 @@ export class ProcBot {
 
 	/**
 	 * Add a new type of ServiceListener to the client.
-	 * Should the ServiceListener already exist on the client, this will do nothing.
+	 * Should the ServiceListener already exist on the client with the handle/name, this will do nothing.
+	 *
 	 * @param name   The name of the ServiceListener to add.
-	 * @param data?  Any relevant data required to construct the ServiceListener.
+	 * @param data?  A ServiceBase extended constructor, should data be passed to a Service constructor.
 	 * @return       The constructed ServiceListener or void should it already exist or fail.
 	 */
 	protected addServiceListener(name: string, data?: any): ServiceListener | void {
-		const service = this.getService(name);
-		let listener;
-
-		if (service && !_.find(this.listeners, [ 'serviceName', name ])) {
-			listener = service.createServiceListener(data);
-			this.listeners.push(listener);
-		}
-
-		return listener;
+		return <ServiceListener>this.addService('listener', name, data);
```

Comment:
Completely preference-based, but I think this could be more readable as
```typescript
return this.addService('listener', name, data) as ServiceListener
```
and again lower down.